dispatcher/main.go

```diff
@@ -0,0 +1,480 @@
+		log.Fatalf(fmt.Errorf(templateError, "http failure", err).Error())
+	}
+
+	var run models.Run
+	err = json.Unmarshal(content, &run)
+	if err != nil {
+		log.Fatalf(fmt.Errorf(templateError, "json unmarshal failure", err).Error())
+	}
+
+	info(fmt.Sprintf("Find run %d: %s.", run.ID, run.Name))
+	if run.Details != nil {
+		info("  Details")
+		for key, value := range run.Details {
+			info(fmt.Sprintf("    %s = %s", key, value))
+		}
+	}
+	if run.Settings != nil {
+		info("  Settings")
+		for key, value := range run.Settings {
+			info(fmt.Sprintf("    %s = %s", key, value))
+		}
+	}
+
+	info(fmt.Sprintf("Update run product in details to %s.", droidMetadata.Product))
+	run.Details[common.KeyProduct] = droidMetadata.Product
+	_, err = sendRequest(http.MethodPatch, fmt.Sprintf("run/%d", runID), run, templateError)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	return &run
+}
+
+func postTasks(run *models.Run, settings []models.TaskSetting) (err error) {
+	info(fmt.Sprintf("To schedule %d tests.", len(settings)))
+	err = nil
+	tasks := make([]models.Task, len(settings))
+
+	for idx, setting := range settings {
+		var task models.Task
+		task.Name = fmt.Sprintf("Test: %s", setting.Classifier["identifier"])
+		task.Settings = setting
+		task.Annotation = run.Settings[common.KeyImageName].(string)
+
+		tasks[idx] = task
+	}
+
+	templateError := fmt.Sprintf("Fail to create task for run %d.", run.ID) + " Reason %s. Exception %s."
+	info("Posting tasks ...")
+	_, err = sendRequest("POST", fmt.Sprintf("run/%d/tasks", run.ID), tasks, templateError)
+	info("Finish posting tasks ...")
+
+	return
+}
+
+func createTaskJob(run *models.Run) (job *batchv1.Job, err error) {
+	client, err := kubeutils.CreateKubeClientset()
+	if err != nil {
+		return nil, err
+	}
+
+	parallelism := int32(run.Settings[common.KeyInitParallelism].(float64))
+	var backoff int32 = 5
+
+	name := fmt.Sprintf("%s-%d-%s", droidMetadata.Product, run.ID, getRandomString())
+	definition := batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: batchv1.JobSpec{
+			Parallelism:  &parallelism,
+			BackoffLimit: &backoff,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+				},
+				Spec: corev1.PodSpec{
+					Containers:       getContainerSpecs(run),
+					ImagePullSecrets: getImagePullSource(run),
+					Volumes:          getVolumes(run),
+					RestartPolicy:    corev1.RestartPolicyNever,
+				},
+			},
+		},
+	}
+
+	return client.BatchV1().Jobs(namespace).Create(&definition)
+}
+
+func getLabels(run *models.Run) map[string]string {
+	labels := make(map[string]string)
+	labels["run_id"] = strconv.Itoa(run.ID)
+	labels["run_live"] = run.Settings[common.KeyLiveMode].(string)
+
+	return labels
+}
+
+func getVolumes(run *models.Run) []corev1.Volume {
+	if droidMetadata.Storage {
+		azureSource := corev1.AzureFileVolumeSource{
+			SecretName: getSecretName(run),
+			ShareName:  run.Settings[common.KeyStorageShare].(string),
```

Comment:
>run.Settings[common.KeyStorageShare].(string) [](start = 15, length = 45)

Personally, I would try to avoid doing type assertions, but feel free to leave it like this :)

Reply:
What's the alternative you would do?

Reply:
Well it is somewhat unavoidable because settings is a `map[string]interface{}`. Personally I prefer  not to create `interface{}` (I like types to be strong), but since the service works that way it would take many changes. If the service remained like it is today, a JSON custom unmarshaller could go well, but taking into account the amount of boilerplate it needs, it is best to forget this comment.

Reply:
Yup. We can have a more rigorous model definition when system settles down a bit.